store/postgres/src/deployment_store.rs

```diff
@@ -704,23 +709,40 @@ impl DeploymentStore {
         entity_name: &str,
         field_names: Vec<String>,
         index_method: Method,
+        after: Option<BlockNumber>,
     ) -> Result<(), StoreError> {
         let store = self.clone();
         let entity_name = entity_name.to_owned();
         self.with_conn(move |conn, _| {
             let schema_name = site.namespace.clone();
             let layout = store.layout(conn, site)?;
             let table = resolve_table_name(&layout, &entity_name)?;
-            let column_names = resolve_column_names(table, &field_names)?;
+            let column_names_to_types = resolve_column_names(table, &field_names)?;
+            let column_names = column_names_to_types
+                .iter()
+                .map(|(name, _)| *name)
+                .collect::<Vec<&SqlName>>();
+
             let column_names_sep_by_underscores = column_names.join("_");
-            let column_names_sep_by_commas = column_names.join(", ");
+            let index_exprs = resolve_index_exprs(column_names_to_types);
             let table_name = &table.name;
             let index_name = format!("manual_{table_name}_{column_names_sep_by_underscores}");
-            let sql = format!(
+            let mut sql = format!(
                 "create index concurrently if not exists {index_name} \
                  on {schema_name}.{table_name} using {index_method} \
-                 ({column_names_sep_by_commas})"
+                 ({index_exprs}) ",
             );
+
+            // If 'after' is provided and the table is not immutable, add a WHERE clause for partial indexing
+            if let Some(after) = after {
+                if !table.immutable {
```

Comment:
Not sure what needs to be done for entities that are immutable? Return an error? or just ignore the partial indexing and log something that suggests --after is being ignored?

Reply:
I think for immutable tables we want to produce an error - these partial indexes are meant to reduce the amount of data the db has to look at to find entities that are visible after block `after`, but immutable entities are visible forever, and such an index doesn't make sense to them. We'd just want `graphman` to provide some error to tell the user that they tried something nonsensical.

Reply:
Ty, Made the change